drivers/rtc/rtc_ll_stm32.c

```diff
@@ -0,0 +1,360 @@
+	LL_EXTI_EnableIT_0_31(LL_EXTI_LINE_18);
+	LL_EXTI_EnableRisingTrig_0_31(LL_EXTI_LINE_18);
+
+	irq_enable(CONFIG_RTC_0_IRQ);
+
+	return 0;
+}
+
+static int rtc_stm32_set_config(struct device *dev, struct rtc_config *cfg)
+{
+	int result = 0;
+	time_t init_ts = 0;
+	struct tm init_tm = { 0 };
+
+	LL_RTC_DateTypeDef rtc_date;
+	LL_RTC_TimeTypeDef rtc_time;
+
+	/* Convert seconds since 2000-01-01 00:00:00 to calendar datetime */
+	init_ts = cfg->init_val;
+	init_ts += EPOCH_OFFSET;
+
+	gmtime_r(&init_ts, &init_tm);
+
+	rtc_date.Year = init_tm.tm_year % 100;
+	rtc_date.Month = init_tm.tm_mon;
+	rtc_date.Day = init_tm.tm_mday;
+
+	rtc_time.TimeFormat = LL_RTC_TIME_FORMAT_AM_OR_24;
+	rtc_time.Hours = init_tm.tm_hour;
+	rtc_time.Minutes = init_tm.tm_min;
+	rtc_time.Seconds = init_tm.tm_sec;
+
+	k_sem_take(DEV_SEM(dev), K_FOREVER);
+
+	if (cfg->cb_fn != NULL) {
+		DEV_DATA(dev)->cb_fn = cfg->cb_fn;
+	}
+
+	if (LL_RTC_DATE_Init(RTC, LL_RTC_FORMAT_BIN, &rtc_date) != SUCCESS) {
+		result = -EIO;
+		goto fin;
+	}
+
+	if (LL_RTC_TIME_Init(RTC, LL_RTC_FORMAT_BIN, &rtc_time) != SUCCESS) {
+		result = -EIO;
+		goto fin;
+	}
+
+	if (cfg->alarm_enable) {
+		rtc_stm32_set_alarm(dev, cfg->alarm_val);
+	}
+
+fin:
+	k_sem_give(DEV_SEM(dev));
+
+	return result;
+}
+
+static u32_t rtc_stm32_get_pending_int(struct device *dev)
+{
+	return LL_RTC_IsActiveFlag_ALRA(RTC) != 0;
+}
+
+void rtc_stm32_isr(void *arg)
+{
+	struct device *const dev = (struct device *)arg;
+
+	if (LL_RTC_IsActiveFlag_ALRA(RTC) != 0) {
+
+		if (DEV_DATA(dev)->cb_fn != NULL) {
+			DEV_DATA(dev)->cb_fn(dev);
+		}
+
+		LL_RTC_ClearFlag_ALRA(RTC);
+		LL_RTC_DisableIT_ALRA(RTC);
+
+	}
+
+	LL_EXTI_ClearFlag_0_31(LL_EXTI_LINE_18);
+}
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+
+#ifdef CONFIG_SYS_POWER_DEEP_SLEEP
+static int rtc_suspend_device(struct device *dev)
+{
+	rtc_stm32_disable(dev);
+	rtc_stm32_set_power_state(dev, DEVICE_PM_SUSPEND_STATE);
+
+	return 0;
+}
+
+static int rtc_resume_device(struct device *dev)
+{
+	rtc_stm32_enable(dev);
+	rtc_stm32_set_power_state(dev, DEVICE_PM_ACTIVE_STATE);
+
+	return 0;
+}
+#endif
+
+/*
+ * Implements the driver control management functionality
+ * the *context may include IN data or/and OUT data
+ */
+static int rtc_stm32_device_ctrl(struct device *dev, u32_t ctrl_command,
+				 void *context)
+{
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+#ifdef CONFIG_SYS_POWER_DEEP_SLEEP
+		if (*((u32_t *)context) == DEVICE_PM_SUSPEND_STATE) {
+			return rtc_suspend_device(dev);
+		} else if (*((u32_t *)context) == DEVICE_PM_ACTIVE_STATE) {
+			return rtc_resume_device(dev);
+		}
+#endif
+	} else if (ctrl_command == DEVICE_PM_GET_POWER_STATE) {
+		*((u32_t *)context) = rtc_stm32_get_power_state(dev);
+		return 0;
+	}
+
+	return 0;
+}
+#endif
+
+static int rtc_stm32_init(struct device *dev)
+{
+	static const struct stm32_pclken pclken = {
+		.enr = LL_APB1_GRP1_PERIPH_PWR,
+		.bus = STM32_CLOCK_BUS_APB1,
+	};
+
+	LL_RTC_InitTypeDef rtc_config = {
```

Comment:
Same here, add this in driver config struct

Reply:
The LL API won't take `const` here, but I'll try